Add configuration file for endeavor

cmcantalupo opened the pull request:
- Do not specify FI_PROVIDER in endeavor config
  since the system has multiple fabrics.
- One work around is to see which lustre file systems
  are loaded but this inference is not very direct.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add configuration file for endeavor

- Do not specify FI_PROVIDER in endeavor config
  since the system has multiple fabrics.
- One work around is to see which lustre file systems
  are loaded but this inference is not very direct.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented on integration/config/endeavor_env.sh:
```diff
@@ -0,0 +1,54 @@
+export MPICXX=mpiicpc
+export FC=ifort
+export F77=ifort
+export F90=ifort
```
Should F90 be added to build_env.sh?

bgeltz commented on integration/config/endeavor_env.sh:
```diff
@@ -0,0 +1,54 @@
+export MPIFORT=mpiifort
+export MPIFC=mpiifort
+export MPIF77=mpiifort
+export MPIF90=mpiifort
```
I think we may want to add a default for MPIF90 to build_env as well.

bgeltz commented on integration/config/endeavor_env.sh:
```diff
@@ -0,0 +1,54 @@
+source /opt/intel/impi/latest/compilers_and_libraries/linux/mpi/intel64/bin/mpivars.sh
+
+export GEOPM_LAUNCHER=impi
+export CC=icc
+export CXX=icpc
```
These are the defaults in build_env.sh. This override is unnecessary.

bgeltz commented on integration/config/endeavor_env.sh:
```diff
@@ -0,0 +1,54 @@
+export CXX=icpc
+export MPICC=mpiicc
+export MPICXX=mpiicpc
+export FC=ifort
+export F77=ifort
```
These are the defaults in build_env.sh. This override is unnecessary.

cmcantalupo committed:
Address review feedback

- Avoid setting variable that are set by default.
- Add a few more names for the fortran compiler.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

bgeltz commented on integration/config/build_env.sh:
```diff
@@ -83,4 +86,3 @@ export GEOPM_FFLAGS="-I${GEOPM_LIB}/${FC}/modules/geopm-x86_64"
 export GEOPM_LDFLAGS="-L${GEOPM_LIB}"
 export GEOPM_LDLIBS="-lgeopm"
 export GEOPM_FORTRAN_LDLIBS="${GEOPM_LDLIBS} -lgeopmfort"
-
```
I will miss this blank line... :)
